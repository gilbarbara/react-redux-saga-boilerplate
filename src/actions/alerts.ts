import { ReactNode } from 'react';
import { uuid } from '@gilbarbara/helpers';
import { createAction } from '@reduxjs/toolkit';

import { actionPayload } from '~/modules/helpers';

import { ActionTypes } from '~/literals';

import { ShowAlertOptions } from '~/types';

export const hideAlert = createAction<string>(ActionTypes.ALERTS_HIDE);

export const showAlert = createAction(
  ActionTypes.ALERTS_SHOW,
  (message: ReactNode, options: ShowAlertOptions) => {
    const timeout = options.type === 'error' ? 0 : 5;

    return actionPayload({
      id: options.id || uuid(),
      icon: options.icon || 'info-o',
      message,
      position: options.position || 'bottom-right',
      type: options.type || 'neutral',
      timeout: typeof options.timeout === 'number' ? options.timeout : timeout,
    });
  },
);
