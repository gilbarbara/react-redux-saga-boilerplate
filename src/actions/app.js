// @flow
/**
 * @module Actions/App
 * @desc App Actions
 */

import uuid from 'uuid/v4';
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export { goBack, go, push, replace } from 'modules/history';

export const { hideAlert, showAlert, switchMenu } = createActions({
  [ActionTypes.SWITCH_MENU]: (query: string) => ({ query }),
  [ActionTypes.HIDE_ALERT]: (id: string) => ({ id }),
  [ActionTypes.SHOW_ALERT]: (message: string, options: Object) => {
    const timeout = options.variant === 'danger' ? 0 : 5;

    return {
      id: options.id || uuid(),
      icon: options.icon,
      message,
      position: options.position || 'bottom-right',
      variant: options.variant || 'dark',
      timeout: typeof options.timeout === 'number' ? options.timeout : timeout,
    };
  },
});
