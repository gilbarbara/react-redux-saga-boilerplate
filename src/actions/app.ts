/**
 * @module Actions/App
 * @desc App Actions
 */
import React from 'react';
import { nanoid } from 'nanoid';

import { createAction } from 'modules/helpers';

import { ActionTypes } from 'literals';

import { ShowAlertOptions } from 'types';

export { go, goBack, push, replace } from 'modules/history';

export const hideAlert = createAction(ActionTypes.HIDE_ALERT, (id: string) => ({ id }));

export const showAlert = createAction(
  ActionTypes.SHOW_ALERT,
  (message: React.ReactNode, options: ShowAlertOptions) => {
    const timeout = options.variant === 'danger' ? 0 : 5;

    return {
      id: options.id || nanoid(),
      icon: options.icon,
      message,
      position: options.position || 'bottom-right',
      variant: options.variant || 'dark',
      timeout: typeof options.timeout === 'number' ? options.timeout : timeout,
    };
  },
);
