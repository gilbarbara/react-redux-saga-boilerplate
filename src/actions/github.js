// @flow
/**
 * @module Actions/User
 * @desc User Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const { githubGetRepos: getRepos } = createActions({
  [ActionTypes.GITHUB_GET_REPOS]: (query: string) => ({ query }),
});
