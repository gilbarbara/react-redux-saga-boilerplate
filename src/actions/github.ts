/**
 * @module Actions/User
 * @desc User Actions
 */
import { createAction } from 'modules/helpers';

import { ActionTypes } from 'literals';

export const getRepos = createAction(ActionTypes.GITHUB_GET_REPOS_REQUEST, (query: string) => ({
  query,
}));
