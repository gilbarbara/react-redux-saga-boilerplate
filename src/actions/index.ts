export * from './app';
export * from './github';
export * from './user';
