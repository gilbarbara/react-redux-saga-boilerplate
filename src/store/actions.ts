export { alertHide, alertShow } from './slices/alerts';
export { setAppOptions } from './slices/app';
export { getRepos, getReposFailure, getReposSuccess } from './slices/github';
export { login, loginSuccess, logOut, logOutSuccess } from './slices/user';
