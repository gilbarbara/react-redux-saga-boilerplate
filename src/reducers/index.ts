import app, { appState } from './app';
import github, { githubState } from './github';
import user, { userState } from './user';

export const initialState = {
  app: appState,
  github: githubState,
  user: userState,
};

export default {
  ...app,
  ...github,
  ...user,
};
