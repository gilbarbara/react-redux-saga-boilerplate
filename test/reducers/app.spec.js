import reducers from 'reducers';
import * as Actions from 'actions';
import { ActionTypes } from 'constants/index';

describe('App', () => {
  it('should return the initial state', () => {
    expect(reducers.app(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.SHOW_ALERT}`, () => {
    expect(reducers.app(undefined, Actions.showAlert('success', 'hello')).notifications).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.HIDE_ALERT}`, () => {
    expect(reducers.app(undefined, Actions.hideAlert()).notifications).toMatchSnapshot();
  });
});
