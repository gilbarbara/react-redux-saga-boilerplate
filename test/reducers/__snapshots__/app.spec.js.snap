// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should handle HIDE_ALERT 1`] = `
Object {
  "message": "",
  "status": "",
  "visible": false,
  "withTimeout": true,
}
`;

exports[`App should handle SHOW_ALERT 1`] = `
Object {
  "message": "hello",
  "status": "success",
  "visible": true,
  "withTimeout": true,
}
`;

exports[`App should return the initial state 1`] = `
Object {
  "notifications": Object {
    "message": "",
    "status": "",
    "visible": false,
    "withTimeout": true,
  },
}
`;
